Apps/Sandcastle/gallery/Callback Property.html

```diff
@@ -0,0 +1,88 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no">
+    <meta name="description" content="Use Viewer to start building new applications or easily embed Cesium into existing applications.">
+    <meta name="cesium-sandcastle-labels" content="Beginner, Showcases">
+    <title>Cesium Demo</title>
+    <script type="text/javascript" src="../Sandcastle-header.js"></script>
+    <script type="text/javascript" src="../../../ThirdParty/requirejs-2.1.20/require.js"></script>
+    <script type="text/javascript">
+    require.config({
+        baseUrl : '../../../Source',
+        waitSeconds : 60
+    });
+    </script>
+</head>
+<body class="sandcastle-loading" data-sandcastle-bucket="bucket-requirejs.html">
+<style>
+    @import url(../templates/bucket.css);
+</style>
+<div id="cesiumContainer" class="fullSize"></div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar"></div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+// This example illustrates a Callback Property, a property whose
+// value is lazily evaluated by a callback function.
+var viewer = new Cesium.Viewer('cesiumContainer');
+
+var startLat = 35;
+var startLon = -120;
+var endLon = startLon;
+
+// Update the polyline endpoints to animate the line length.
+function getPositions() {
+    endLon += 0.01;
+    return Cesium.Cartesian3.fromDegreesArray([startLon, startLat, endLon, startLat]);
+}
+
+// Add a polyline to the scene. Positions are dynamic.
+var redLine =  viewer.entities.add({
+    polyline : {
+        // This callback updates positions each frame.
+        positions : new Cesium.CallbackProperty(getPositions, false),
+        width : 5,
+        material : Cesium.Color.RED
+    }
+});
+
+var startCarto = Cesium.Cartographic.fromDegrees(startLon, startLat);
+var endCarto = new Cesium.Cartographic(); // use scratch object to avoid new allocations per frame.
+var geodesic = new Cesium.EllipsoidGeodesic();
+
+// Calculate the length of the line
+function getLength() {
+    // Get the end position from the polyLine's callback.
+    var endPoint = redLine.polyline.positions.getValue()[1];
+    endCarto = Cesium.Cartographic.fromCartesian(endPoint);
+    
+    geodesic.setEndPoints(startCarto, endCarto);
+    return 'Length: ' + Math.round(geodesic.surfaceDistance);
+}
+
+// Label the polyline with calculated length.
+var label = viewer.entities.add({
+    position : Cesium.Cartesian3.fromDegrees(startLon, startLat),
```

Comment:
Why not compute the centerpoint of the line and place the label there?  Seems like another good use of the callback property?  You can actually have this callback property call the other callback property to get it's points.